src/fundus/scraping/url.py

```diff
@@ -13,19 +14,51 @@
 from requests import ConnectionError, HTTPError
 
 from fundus.logging import create_logger
+from fundus.parser.utility import generic_nodes_to_text
 from fundus.scraping.filter import URLFilter, inverse
 from fundus.scraping.session import _default_header, session_handler
 
 logger = create_logger(__name__)
 
 
+class CompressionFormats(Enum):
```

Comment:
From what I see, `bz2`, `zip` and `lzma` would raise a `NotImplementedError`. Without having any background knowledge it seems odd to me that you specifically chose these four compression formats and only implemented the decompression for one of them.

Reply:
I asked ChatGPT about the most common compression formats and added them as templates.

Update: I added the missing formats (except ZIP) as I have no clue how to handle ZIP archives unless we stumble upon one.